lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2341,20 +2360,17 @@ class AvailabilityWalker : public ASTWalker {
   /// Walk a member reference expression, checking for availability.
   void walkMemberRef(MemberRefExpr *E) {
     // Walk the base in a getter context.
+    // FIXME: We may need to look at the setter too, if we're going to do
+    // writeback. The AST should have this information.
```

Comment:
We could do something silly here, where we keep track of the operands of `InOutExpr` nodes and if `E` is in that set, we also check the setter.

Reply:
I want to use LValueAccessKind if Slava doesn't take it out first! But in a separate PR.